src/plot.js

```diff
@@ -171,32 +177,62 @@ export function plot(options = {}) {
     const indexByFacet = facetMap(facetChannels);
     facets.forEach(([key], i) => indexByFacet.set(key, i));
     const selection = select(svg);
-    if (fy && axes.y) {
-      const axis1 = axes.y, axis2 = nolabel(axis1);
-      const j = axis1.labelAnchor === "bottom" ? fyDomain.length - 1 : axis1.labelAnchor === "center" ? fyDomain.length >> 1 : 0;
-      selection.selectAll()
-        .data(fyDomain)
-        .enter()
-        .append((ky, i) => (i === j ? axis1 : axis2).render(
-          fx && where(fxDomain, kx => indexByFacet.has([kx, ky])),
-          scales,
-          {...dimensions, ...fyMargins, offsetTop: fy(ky)},
-          context
-        ));
+    if (fy) {
+      if (options.fy?.grid === true) {
+        const ty = fy.bandwidth() / 2;
+        selection.selectAll()
+          .data(fyDomain.map(d => fy(d) + ty))
+          .enter()
+            .append("line")
+            .attr("y1", d => d)
+            .attr("y2", d => d)
+            .attr("x1", dimensions.marginLeft)
+            .attr("x2", dimensions.width - dimensions.marginRight)
+            .attr("stroke", "currentColor")
+            .attr("stroke-opacity", 0.1);
+      }
```

Comment:
Having a (non-configurable) alternate path for the grid on the facet axes seems unfortunate. Like, if Plot supports grid on _fx_ and _fy_, then I’d want to be able to use the Plot.grid mark (or Plot.gridFX? Plot.facetGrid? Plot.frameGrid?) to control how that grid is rendered. I guess that’s not possible because the facet grid needs to extend _across_ facet frames, and thus can’t live in the same space as other marks?

I wonder if there’s a more configurable way that we could do this. One option would be to have a “frame” grid that is repeated for each frame (changing the visual appearance of facet grids). Another option would be to have an array of “facet marks” (name TBD) that are rendered separately below (or above?) the facet frames in a single pass, perhaps where _x_ = _fx_ and _y_ = _fy_.

I think eventually it’d be nice to figure out how to replace the built-in axes with a custom axis mark, for example…

Reply:
I'm not sure about the use case for the facet grid (it was there so I tried to keep it, and felt it was better to have it outside of axes). I like the idea of having a mark where x would be fx (and y fy); maybe some marks would work out of the box (like Plot.cell) if called in this particular context.

> replace the built-in axes with a custom axis mark

Yes, it seems like a natural fast-follow to this PR. It would also help a bit with #147.